### PR TITLE
Add Radarr package

### PR DESCRIPTION
**Pull request by ressu**
_Motivation:_ Add package for Radarr, a movie downloader based on Sonarr
_Linked issues:_ #2602 

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by ikkeenjij36** [comment]
Second this

**Comment by foXaCe** [comment]
work great thank

**Comment by SuperDOS** [comment]
Is this released? Is there a packet to download?

**Comment by meffect** [comment]
this exists?? awesome!

**Comment by frossmant** [comment]
Great job, here are my findings of trying to compile your package.

--- Building form spksrc ----
Hmm i tried to build the package myself from spksrc.
The build worked okey besides that the spksrc OpenSSL package points to a broken FTP link.

to solve the openssl issue during build i've just edited /spksrc/cross/openssl/Makefile:
#PKG_DIST_SITE = ftp://ftp.openssl.org/source
PKG_DIST_SITE = https://www.openssl.org/source/old/1.0.2

Now it seems to build correctly from what i've seen.

--- Installing .spk package from "synology app center" ----

Works as expected, installs without any problems

--- Starting / Stopping ----

* The Radarr package won't start using the "Synology App Center"
when starting from the synology web gui i got this error in the radarr logfile 
( /usr/local/radarr/var/.config/Radarr/logs/radarr.txt)
http://pastebin.com/rKsjcF2T

* Starting Radarr from terminal
The normal "/var/packages/radarr/scripts/start-stop-status start" init script works as expected the same as doing it from the synology web ui, just crashes upon startup.

I've tried to start radarr manually from it's own user and that makes Radarr start.

$ su - radarr
$ /usr/local/mono/bin/mono --debug /usr/local/radarr/share/Radarr/Radarr.exe

Here's the startup logfile, http://pastebin.com/0rK70wWn

* MediaInfo
When starting the package manually from it's radarr user the app starts up as it should but i get the following issue in the "system/status" page in the webui of radarr.

"MediaInfo could not be loaded libmediainfo.so.0"

Maybe some enviroment path problem or something ? I've just tried to recompile libmediainfo from spksrc and it seems to compile as it should.

**Comment by ressu** [comment]
There has been a few releases of Radarr after the package was created. One of the fixes in the future is related to certain libraries. So that issue is most likely been fixed upstream already.

As for the startup issue, it could be related to this: #2216 

Also the extremely interesting thing in your log is that Radarr is starting with port 5002, which isn't the default port for Radarr.

**Comment by frossmant** [comment]
@ressu don't mind the port, that is my custom configuration :)

**Comment by frossmant** [comment]
@ressu yeh could be some issues with radar itself related to the library, i've updated it to the latest version yesterday though and that did not solve the issue. 

Could not manage to get the internal updating function to work on my synology though.

**Comment by ressu** [comment]
Interesting, my instance is happily updating consistently. Can you check the logs to see what the actual error is? (whether we are missing a dependency or if it's a permission error)

**Comment by ressu** [comment]
Rebased the branch on current master and bumped Radarr version to the latest Radarr release.

**Comment by thekingsky** [comment]
Looking forward to seeing this packet on synocommunity :)

**Comment by KaraokeStu** [comment]
LGTM, @cytec Can you confirm you're happy with it now?

**Comment by Dr-Bean** [comment]
Please replace occurrences of `SONARR` and `nzbdrone` with `RADARR` and `radarr` respectively (in the actual code, not necessarily in package descriptions/comments etc).

**Comment by ressu** [comment]
Oh, I did apparently miss a few. Will fix tonight and update the PR.

File: spk/radarr/PLIST [hunk 1]
```diff
@@ -0,0 +1,3 @@
+lnk:lib/libsqlite3.so
```
Comment (Dr-Bean) [inline spk/radarr/PLIST]: This doesn't seem correct, as it's already included via `cross/sqlite/PLIST`. Please remove this, we'll do the same for the Sonarr package (or maybe you can create a PR for that as well)

File: spk/radarr/src/app/config [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    ".url": {
+        "com.synocommunity.packages.radarr": {
+            "title": "Radarr",
+            "desc": "Radarr Web UI",
+            "icon": "images/radarr-{0}.png",
+            "type": "url",
+            "protocol": "http",
+            "port": "8988",
```
Comment (Dr-Bean) [inline spk/radarr/src/app/config]: This is not the correct port.

File: spk/radarr/src/dsm-control.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+# Package
+PACKAGE="radarr"
+DNAME="Radarr"
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+PATH="${INSTALL_DIR}/bin:${PATH}"
+USER="${PACKAGE}"
+USER_HOME="$(eval echo ~$USER)"
```
Comment (Dr-Bean) [inline spk/radarr/src/dsm-control.sh]: Please remove (so I don't have to do this afterwards ;)). Where `USER_HOME` is used, please use `${INSTALL_DIR}/var` instead

File: spk/radarr/Makefile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+DSM_UI_DIR = app
+
+REQUIRED_DSM = 5.0
+UNSUPPORTED_ARCHS = $(PPC_ARCHES)
+
+DEPENDS = cross/busybox cross/curl cross/mediainfo cross/sqlite cross/radarr
+
+SPK_DEPENDS = "mono>3.6"
+
+MAINTAINER = SynoCommunity
```
Comment (Dr-Bean) [inline spk/radarr/Makefile]: You are the maintainer of this package :)

**Comment by Dr-Bean** [comment]
I've added some more comments, please take a look.

File: spk/radarr/PLIST (hunk 1 above)
Comment (Dr-Bean) [inline spk/radarr/PLIST]: Opened https://github.com/SynoCommunity/spksrc/issues/2632 for Sonarr.

**Comment by Dr-Bean** [comment]
Almost forgot...Please add `BETA = 1` to the SPK's Makefile.

File: spk/radarr/PLIST (hunk 1 above)
Comment (ressu) [inline spk/radarr/PLIST]: Is it OK if I remove the whole file, or does it have to exist?

**Comment by ressu** [comment]
There we go, let me know if it looks good and I'll squash the commits

**Comment by ressu** [comment]
In that case, squashed commits into a single commit.

**Comment by KaraokeStu** [comment]
Have you tested it still all builds successfully?

**Comment by ressu** [comment]
all-supported completes normally and I'm just in the process of installing it to my NAS

File: spk/radarr/src/installer.sh [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+INSTALL_LOG="${INSTALL_DIR}/var/install.log"
+TMP_INSTALL_LOG="${TMP_DIR}/${PACKAGE}/var/install.log"
+SSS="/var/packages/${PACKAGE}/scripts/start-stop-status"
+PATH="${INSTALL_DIR}/bin:${PATH}"
+USER="${PACKAGE}"
+GROUP="users"
+PID_FILE="${INSTALL_DIR}/var/.config/${DNAME}/nzbdrone.pid"
+MONO_PATH="/usr/local/mono/bin"
+MONO="${MONO_PATH}/mono"
+SONARR="${INSTALL_DIR}/share/${DNAME}/Radarr.exe"
```
Comment (Dr-Bean) [inline spk/radarr/src/installer.sh]: `SONARR`?

File: spk/radarr/src/installer.sh [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+SPK_SONARR="${SYNOPKG_PKGINST_TEMP_DIR}/share/${DNAME}/Radarr.exe"
```
Comment (Dr-Bean) [inline spk/radarr/src/installer.sh]: `SONARR`?

File: spk/radarr/src/installer.sh [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    # Stop the package
+    ${SSS} stop > /dev/null
+
+    # Save some stuff
+    rm -fr ${TMP_DIR}/${PACKAGE}
+    mkdir -p ${TMP_DIR}/${PACKAGE}
+    mv ${INSTALL_DIR}/var ${TMP_DIR}/${PACKAGE}/
+
+    # Is Installed Radarr Binary Ver. >= SPK Radarr Binary Ver.?
+    CUR_VER=$(${COMMAND} ${SONARR} --? | grep -o "Version.*" | awk '{print $2;exit;}' | tr -d '.')
```
Comment (Dr-Bean) [inline spk/radarr/src/installer.sh]: `SONARR`?

File: spk/radarr/src/installer.sh [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+    echo "   Installed Radarr Binary: ${CUR_VER}" >> ${TMP_INSTALL_LOG}
+    SPK_VER=$(${COMMAND} ${SPK_SONARR} --? | grep -o "Version.*" | awk '{print $2;exit;}' | tr -d '.')
```
Comment (Dr-Bean) [inline spk/radarr/src/installer.sh]: `SONARR`?

File: spk/radarr/src/installer.sh (hunk 5 above)
Comment (ressu) [inline spk/radarr/src/installer.sh]: Argh.. I though I found all these.. Fixing my blindess.

Comment (Dr-Bean) [inline spk/radarr/src/installer.sh]: 👍 ;)

File: spk/radarr/PLIST (hunk 1 above)
Comment (Dr-Bean) [inline spk/radarr/PLIST]: We usually keep the file, but empty.

**Comment by ressu** [comment]
There we go.. At least `git grep -i sonarr` doesn't find anything but the description anymore

**Comment by ressu** [comment]
There we go and squashed the commits again for a clean merge.

**Comment by SuperDOS** [comment]
complied it for cedarview and works great on my 1812+ 💯 
great work!

**Comment by newkind** [comment]
Any ideas when this will appear in Packages Manager? :)

**Comment by SuperDOS** [comment]
@newkind 
it's quite easy to compile it yourself with the instructions here if you don't want to wait :) https://github.com/SynoCommunity/spksrc/wiki/Developers-HOW-TO

I compiled it yesterday for cedarview and run it on my 1812+
Attach it here for anyone that wants it.

Works great.

Thanks everyone that helped bring radarr to synology!
[radarr_cedarview-5.2_20170204-1.zip](https://github.com/SynoCommunity/spksrc/files/752874/radarr_cedarview-5.2_20170204-1.zip)

**Comment by newkind** [comment]
@SuperDOS Thanks, I compiled it myself ;) Took a little bit longer as it was my initial compile but works like it should!

**Comment by oliroe** [comment]
Hi. Any update on when this will be available in package manager?

Thanks

**Comment by KaraokeStu** [comment]
Just realised to package manager :)

**Comment by KaraokeStu** [comment]
Or released... Lol

**Comment by oliroe** [comment]
Awesome, installing now. Great work guys

**Comment by MostDefiantly** [comment]
I tried installing, but the package fails saying port 7878 is reserved for system use or in use by another service. I verified there is nothing responding on that port. 

DS415+ running DSM 5.2-5967 Update 2

**Comment by oliroe** [comment]
I'm on DSM 6 on a DS213 so at least that eliminates the DSM version from being the problem

**Comment by oliroe** [comment]
It's working when I disable the Synology Firewall

**Comment by MostDefiantly** [comment]
Looking at the firewall, I was able to determine that ports 7400-7999 are system reserved for High Availability. I am not using HA at all so the port is free - but the OS itself is blocking the package install since it's in a system range. 

I would suggest perhaps moving the default port for this package outside any system reserved ranges?

**Comment by zebulon501** [comment]
Works fine for me, thanks for the efforts!

DS1812+ DSM 6.0.2-8451 Update 9

**Comment by oliroe** [comment]
Rebooted my NAS and now the package won't start in any combination of Radarr, Sonarr and the Firewall. Something definitely not right here

**Comment by KaraokeStu** [comment]
I'll get this tested further tomorrow on my clean vms but currently running this as stock on my dsm 6 update 9 vm

**Comment by KaraokeStu** [comment]
I have both working, but have switched the port I run it on, as I like to have my services sequential from 8080

**Comment by MostDefiantly** [comment]
Perhaps DSM5 treats the HA ports differently from DSM6. I feel moving it to the 8000's may be the cleanest bet: https://github.com/SynoCommunity/spksrc/wiki/SynoCommunity-Used-Ports

Here's the port range reserved for HA in DSM5:

image

**Comment by ressu** [comment]
I'm a bit conflicted on the fix. I could change the port from 7878 to something like 8878, but that would make the port less predictable for those who already know the default port of Radarr. At the same time I know that we have to switch the default port.

**Comment by Dr-Bean** [comment]
Switching to another port seems the correct approach to me :)
The standard approach that we have for SynoCommunity is that if the default port isn't available, we switch to another (and choose a port as per the wiki instructions). Even though HA isn't available on all models, we can't have a package working on some model -- but not on others, and we can't have a package that uses a different port based on the model either. We have multiple packages that don't use the default port due to conflicts, no problem.

@MostDefiantly (or maybe someone else): is that set of ports visible on the Synology website somewhere? It doesn't look like it's on the Synology port overview here: https://www.synology.com/en-global/knowledgebase/DSM/tutorial/General/What_network_ports_are_used_by_Synology_services. We should probably ask Synology to update that page otherwise.

**Comment by MostDefiantly** [comment]
@Dr-Bean  Oddly, I noticed that it was missing as well when researching yesterday. Looks like there's some gaps there, as HA isn't even mentioned in their matrix.

I could snapshot all the service ports that are possible to add the the firewall if it helps.

**Comment by Dr-Bean** [comment]
@MostDefiantly Sure, that'd be nice. Screenshot would be helpful, adding them directly to https://github.com/SynoCommunity/spksrc/wiki/Synology-Used-Ports would be best.

**Comment by MostDefiantly** [comment]
Oh duh, hehe. I've added what I found missing.

**Comment by madvtak** [comment]
Hi @ressu can you take a look at this?

https://github.com/Radarr/Radarr/issues/654

**Comment by KaraokeStu** [comment]
Hi @madvtak - could you open a new issue for that bug, it needs its own, thanks!

**Comment by madvtak** [comment]
Done, thank you @KaraokeStu

**Comment by MostDefiantly** [comment]
@ressu Have you decided on your port yet? *nudge nudge* :)

**Comment by ressu** [comment]
The new ports will be 8310 and 8311 (marked down to wiki a while back) I'll try and get a PR done this weekend

**Comment by MostDefiantly** [comment]
Awesome. I look forward to this weekend. :)

**Comment by ressu** [comment]
The port issue is being tracked here: #2652

**Pull request merged**